internal/pkg/funccalls/analyzer.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package funccalls contains an analyzer that performs identification of
+// sink and sanitizer function definitions and calls. It also identifies
+// excluded functions.
+package funccalls
+
+import (
+	"github.com/google/go-flow-levee/internal/pkg/config"
+	"github.com/google/go-flow-levee/internal/pkg/utils"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/buildssa"
+	"golang.org/x/tools/go/ssa"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "funccalls",
+	Run:      runTest,
+	Flags:    config.FlagSet,
+	Doc:      `The funccalls analyzer finds calls to sink and sanitizer functions.`,
+	Requires: []*analysis.Analyzer{buildssa.Analyzer},
+}
+
+func runTest(pass *analysis.Pass) (interface{}, error) {
+	in := pass.ResultOf[buildssa.Analyzer].(*buildssa.SSA)
+
+	conf, err := config.ReadConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, f := range in.SrcFuncs {
+		if conf.IsExcluded(utils.DecomposeFunction(f)) {
+			continue
+		}
+		for _, b := range f.Blocks {
+			for _, i := range b.Instrs {
+				c, ok := i.(*ssa.Call)
+				if !ok {
+					continue
+				}
+				callee := c.Call.StaticCallee()
+				if callee == nil {
+					continue
+				}
+				switch {
+				case conf.IsSink(utils.DecomposeFunction(callee)):
+					reportCall(pass, c, callee, "sink")
+				case conf.IsSanitizer(utils.DecomposeFunction(callee)):
+					reportCall(pass, c, callee, "sanitizer")
+				}
```

Comment:
Just spitballing here:

Would it be better conceptualization to separate sink and sanitizer identification?  I.e.

```golang
func funcAnalyzer(desc string, matchers []config.FuncMatcher) *analysis.Analyzer {
  // ...
}

var (
  sinkAnalyzer = funcAnalyzer("sink", c.Sinks)
  sanitizerAnalyzer = funcAnalyzer("sanitizer", c.Sanitizers)
)
```

Probably too much gilding for one rose, but it would simplify the analyzer to "just identify this one category."  \shrug

Reply:
I think in terms of how this is used it's better to have a single analyzer.

In #193 the intent is to report all configured calls, which includes both sinks and sanitizers.

In #197 `levee` needs to know which calls are sinks and `source` needs to know which calls are sinks and which are sanitizers.

It seems most clients need to know about both kinds of calls.

Another point is that these two analyzers would have nearly identical code, even down to their tests, indeed they would have nearly the same behavior.